src/lightning/fabric/loggers/tensorboard.py

```diff
@@ -265,6 +265,11 @@ def log_graph(self, model: Module, input_array: Optional[Tensor] = None) -> None
             input_array = model._on_before_batch_transfer(input_array)  # type: ignore[operator]
             input_array = model._apply_batch_transfer_handler(input_array)  # type: ignore[operator]
             self.experiment.add_graph(model, input_array)
+        elif hasattr(model, "module") and callable(getattr(model, "_redirection_through_forward", None)):
```

Comment:
I suggest to do this instead, which should be more robust:

```py
from lightning.fabric.wrappers import _unwrap_objects
model = _unwrap_objects(model)
```

We should probably extend this test here to address the TODO:
https://github.com/Lightning-AI/lightning/blob/66293758c0192955148e8b4bd9dd156927c23503/tests/tests_fabric/loggers/test_tensorboard.py#L149-L151

I can help with that if you wish.